static/images/diagrams/image_workflow.txt

```diff
@@ -0,0 +1,40 @@
+# Use http://sequencediagram.org/ to generate image
+
+title Image-based provisioning with Puppet
+
+participant User
+participant "Provisioned Host" as Host
+participant "Foreman" as Foreman
+participant "Foreman Proxy" as Proxy
+participant "Hypervisor" as HVM
+participant DHCP
+participant DNS
+participant "Puppet master" as Puppet
+
+linear
+autonumber
+
+User->Foreman:clicks on New Host
+User->Foreman:selects virtualization resource and submits
+Foreman->Proxy:requests free IP address
+Proxy->DHCP:forwards free IP request
+Foreman->Proxy:creates DHCP reservation
+Proxy->DHCP:forwards DHCP reservation
+Foreman->Proxy:creates DNS records
+Proxy->DNS:forwards DNS records
+Foreman->Proxy:creates DNS records
+Foreman->HVM:creates new VM
```

Comment:
This happens before creating DHCP reservations, because it first reads the MAC address from the newly created VM.